### ISSUE TITLE
TextInput's textarea component doesn't grow with height of TextInput

### ISSUE THREAD

**Issue by jmbldwn** [callstack/react-native-paper#3673]
### Current behaviour
I set a TextInput component with `style={{flex:1]]` and `multiline`.  The outer bounds of the input grows vertically as expected, but the bounds of the textarea within it are not growing.

### Expected behaviour
If I set the height of the TextInput larger, the multiline textarea should grow with it.

### How to reproduce?
Here's a snack that shows the problem:
https://snack.expo.dev/@jmbldwn/textinput?name=TextInput&description=https%3A%2F%2Fcallstack.github.io%2Freact-native-paper%2Ftext-input.html&code=import%20*%20as%20React%20from%20%27react%27%3B%0Aimport%20%7B%20TextInput%20%7D%20from%20%27react-native-paper%27%3B%0A%0Aconst%20MyComponent%20%3D%20()%20%3D%3E%20%7B%0A%20%20const%20%5Btext%2C%20setText%5D%20%3D%20React.useState(%22%22)%3B%0A%0A%20%20return%20(%0A%20%20%20%20%3CTextInput%0A%20%20%20%20%20%20label%3D%22Email%22%0A%20%20%20%20%20%20value%3D%7Btext%7D%0A%20%20%20%20%20%20onChangeText%3D%7Btext%20%3D%3E%20setText(text)%7D%0A%20%20%20%20%2F%3E%0A%20%20)%3B%0A%7D%3B%0A%0Aexport%20default%20MyComponent%3B&dependencies=react-native-paper,@expo/vector-icons%40%5E13.0.0,react-native-safe-area-context

### Preview
See Snack.

### What have you tried so far?
The stock react native TextInput doesn't have this problem.
I am able to force flex attributes on the textarea in my debugger, but this isn't a solution.

### Your Environment

| software                      | version
| --------------------- | -------
| ios                                | n/a
| android                        | n/a
| react-native                | 0.69.4
| react-native-paper    | 5.2.0
| node                            | 14.16.0
| npm or yarn                | yarn 1.x
| expo sdk                     | 46.0.7

**Comment by tjaniczek** [comment]
Hi @jmbldwn ! So this looks more like a feature request than a bug, since at the moment the field supports only `TextStyle` or `height` : https://callstack.github.io/react-native-paper/text-input.html#style

The reason behind it is that the component itself does a lot of math deciding on where/how to distribute elements on vertical plane. This is not easily transferable to flex layouts. Please note that supporting flexbox is not as simple as covering `flex: 1` case, since there may be multiple edge case scenarios in regards to flex glow, shrink and basis. 

Something to discuss, for sure. But not a bug at the moment in my opinion. Happy to hear feedback from other users/contributors!

<img width="404" alt="image" src="https://user-images.githubusercontent.com/7429558/219616101-97145c69-a0b7-4efc-a948-f47d7327fcf9.png">

**Comment by jmbldwn** [comment]
Agreed, this is probably a feature request, but I think it's not an uncommon use case, especially for mobile.

**Comment by sufyansalim** [comment]
this feature would be really nice to have the textfield having properties of a text area like we have in react to also have in react native.
